neuror/cli.py

```diff
@@ -56,6 +56,51 @@ def repair():
     '''CLI utilities to repair cut planes'''
 
 
+@cli.group()
+def error_annotation():
+    '''CLI utilities related to error annotations'''
+
+
+@error_annotation.command(short_help='Annotate errors on a morphology')
+@click.argument('input_file', type=click.Path(exists=True, file_okay=True))
+@click.argument('output_file')
+@click.option('--error_summary_file', type=click.Path(file_okay=True), default='error_summary.json',
+              help='Path to json file to save error summary')
+def file(input_file, output_file, error_summary_file):
+    from neuror.error_annotation import annotate_single_morphology
+
+    if Path(input_file).suffix not in  ['.asc', '.ASC']:
+        raise Exception('Only .asc/.ASC files are allowed, please convert with morph-tool.')
+
+    annotations, summary = annotate_single_morphology(input_file)
+    shutil.copy(input_file, output_file)
+    with open(output_file, 'a') as morph_file:
+        morph_file.write(annotations)
+    with open(error_summary_file, 'w') as summary_file:
+        json.dump(summary, summary_file)
+
+
+@error_annotation.command(short_help='Annotate errors on morphologies')
+@click.argument('input_dir')
+@click.argument('output_dir', type=click.Path(exists=True, file_okay=False, writable=True))
+@click.option('--error_summary_file', type=click.Path(file_okay=True), default='error_summary.json',
+              help='Path to json file to save error summary')
+def folder(input_dir, output_dir, error_summary_file):
+    from morph_tool.utils import iter_morphology_files
+    from neuror.error_annotation import annotate_morphologies
+
+    output_dir = Path(output_dir)
+    morph_paths = iter_morphology_files(input_dir)
+    annotations, summaries = annotate_morphologies(morph_paths)
+    for morph_path, annotation in annotations.items():
+        output_file  = output_dir / Path(morph_path).name
+        shutil.copy(morph_path, output_file)
```

Comment:
I'm confused about the process here. It first copies the original file in the output directory, and then adds annotations to the file, is this the case?

Reply:
yes, the annotations are just some text appended at the end of the .asc file. I followed what was done in morph-service/view.p.